lib/LibraryTemplatePlugin.js

```diff
@@ -6,30 +6,45 @@
 
 const SetVarMainTemplatePlugin = require("./SetVarMainTemplatePlugin");
 
+/** @typedef {import("./Compiler")} Compiler */
+
+/**
+ * @param {string[]} accessor the accessor to convert to path
+ * @returns {string} the path
+ */
 const accessorToObjectAccess = accessor => {
-	return accessor
-		.map(a => {
-			return `[${JSON.stringify(a)}]`;
-		})
-		.join("");
+	return accessor.map(a => `[${JSON.stringify(a)}]`).join("");
 };
 
-const accessorAccess = (base, accessor, joinWith) => {
-	accessor = [].concat(accessor);
-	return accessor
-		.map((a, idx) => {
-			a = base
-				? base + accessorToObjectAccess(accessor.slice(0, idx + 1))
-				: accessor[0] + accessorToObjectAccess(accessor.slice(1, idx + 1));
-			if (idx === accessor.length - 1) return a;
+/**
+ * @param {string=} base the path prefix
+ * @param {string|string[]} accessor the accessor
```

Comment:
This is where I'd tell you to use `string|ReadonlyArray<string>` too but this would break `Array.isArray`. I recommend adding this to a global `.d.ts`, which will make `Array.isArray` support both `ReadonlyArray` and regular arrays:

```ts
interface ArrayConstructor {
  isArray(arg: any): arg is ReadonlyArray<any>;
}
```

I actually sent this as a PR to TypeScript, but it was rejected on compatibility grounds.